src/main/scala/io/iohk/ethereum/domain/SignedTransaction.scala

```diff
@@ -23,98 +23,94 @@ object SignedTransaction {
-        tx.receivingAddress.toArray,
-        tx.value,
-        tx.payload,
-        Config.Blockchain.chainId,
-        valueForEmptyR,
-        valueForEmptyS)))
-  }
-
-
-  lazy val recoveredSenderAddress: Option[Address] =
-    for {
-      key <- recoveredPublicKey
-      addrBytes = crypto.kec256(key).slice(FirstByteOfAddress, LastByteOfAddress)
-      if addrBytes.length == Address.Length
-    } yield Address(addrBytes)
 
   /**
     * new formula for calculating point sign post EIP 155 adoption
     * v = CHAIN_ID * 2 + 35 or v = CHAIN_ID * 2 + 36
     */
-  lazy val recoveredPointSign: Option[Int] =
+  def getRecoveredPointSign(pointSign: Byte): Option[Int] = {
     if (pointSign == negativePointSign || pointSign == (Blockchain.chainId * 2 + newNegativePointSign).toByte) {
       Some(negativePointSign)
     } else if (pointSign == positivePointSign || pointSign == (Blockchain.chainId * 2 + newPositivePointSign).toByte) {
       Some(positivePointSign)
     } else {
       None
     }
-
-  lazy val recoveredPublicKey: Option[Array[Byte]] = recoveredPointSign.flatMap { p =>
-    ECDSASignature.recoverPubBytes(
-      new BigInteger(1, signatureRandom.toArray[Byte]),
-      new BigInteger(1, signature.toArray[Byte]),
-      ECDSASignature.recIdFromSignatureV(p),
-      bytesToSign
-    )
   }
 
-  lazy val syntacticValidity: Boolean = {
+  def getSender(tx: Transaction, pointSign: Byte, signatureRandom: BigInt, signature: BigInt, recoveredPointSign: Int): Option[Address] = {
+    val bytesToSign: Array[Byte] = if (pointSign == negativePointSign || pointSign == positivePointSign) {
+      //global transaction
+      crypto.kec256(
+        rlpEncode(RLPList(
+          tx.nonce,
+          tx.gasPrice,
+          tx.gasLimit,
+          tx.receivingAddress.toArray,
+          tx.value,
+          tx.payload)))
+    } else {
+      //chain specific transaction
+      crypto.kec256(
+        rlpEncode(RLPList(
+          tx.nonce,
+          tx.gasPrice,
+          tx.gasLimit,
+          tx.receivingAddress.toArray,
+          tx.value,
+          tx.payload,
+          Config.Blockchain.chainId,
+          valueForEmptyR,
+          valueForEmptyS)))
+    }
+
+    val recoveredPublicKey: Option[Array[Byte]] =
+      ECDSASignature.recoverPubBytes(
+        new BigInteger(1, signatureRandom.toByteArray),
+        new BigInteger(1, signature.toByteArray),
+        ECDSASignature.recIdFromSignatureV(recoveredPointSign),
+        bytesToSign
+      )
+
+    for {
+      key <- recoveredPublicKey
+      addrBytes = crypto.kec256(key).slice(FirstByteOfAddress, LastByteOfAddress)
+      if addrBytes.length == Address.Length
+    } yield Address(addrBytes)
+  }
 
-    import tx._
-    import Transaction._
+  def apply(tx: Transaction, pointSign: Byte, signatureRandom: ByteString, signature: ByteString): Option[SignedTransaction] =
+    SignedTransaction(tx, pointSign, BigInt(1, signatureRandom.toArray), BigInt(1, signature.toArray))
```

Comment:
Now that we have this factory method, perhaps it would make more sense if `SignedTransaction` had `signature: ECDSASignature` instead of the individual components?